core/src/main/java/com/yahoo/oak/AbstractSlice.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2020, Verizon Media.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.yahoo.oak;
+
+/* An abstract Slice represents the information about an off-heap cut: a portion of a bigger block,
+** which is part of the underlying managed off-heap memory. Concrete implementation adapts
+** the abstract Slice to a specific memory manager.
+** Slice is allocated only via memory manager, and can be de-allocated later.
+** Slice can be either empty or associated with an off-heap cut,
+** which is the aforementioned portion of an off-heap memory.*/
+abstract class AbstractSlice implements Slice, Comparable<AbstractSlice> {
+    protected static final int UNDEFINED_LENGTH_OR_OFFSET_OR_ADDRESS = -1;
+
+    /** The fields describing the associated off-heap cut, they are set when slice is not empty **/
+    protected long reference;
+    protected int blockID;
+    protected int offset;
+    protected int length; // The entire length of the off-heap cut, including the header (if needed)!
+    protected long memAddress;
+
+    // true if slice is associated with an off-heap slice of memory
+    // if associated is false the Slice is empty
+    protected boolean associated;
+
+    /* ------------------------------------------------------------------------------------
+     * Protected Constructor
+     * ------------------------------------------------------------------------------------*/
+    protected AbstractSlice() {
+        invalidate();
+    }
+
+    /**
+     * Allocate new off-heap cut and associated this slice with a new off-heap cut of memory
+     *
+     * @param size     the number of bytes required by the user
+     * @param existing whether the allocation is for existing off-heap cut moving to the other
+     *                 location (e.g. in order to be enlarged).
+     */
+    public abstract void allocate(int size, boolean existing);
```

Comment:
OK. I removed `Slice`'s interface methods which were `abstract` from `AbstractSlice`.

Regarding the comments I do believe it better to have a "repeated" comment, for the convenience of code inspection. But since I removed all those methods from `AbstractSlice` there are no double comments now.